src/Microsoft.Health.Dicom.Core/Features/Store/StoreDatasetValidator.cs

```diff
@@ -87,9 +87,9 @@ private static void ValidateCoreTags(DicomDataset dicomDataset, string requiredS
                 DicomCoreResource.DuplicatedUidsNotAllowed);
         }
 
-        // If the requestedStudyInstanceUid is specified, then the StudyInstanceUid must match.
+        // If the requestedStudyInstanceUid is specified, then the StudyInstanceUid must match, ignoring whitespace.
         if (requiredStudyInstanceUid != null &&
-            !studyInstanceUid.Equals(requiredStudyInstanceUid, StringComparison.OrdinalIgnoreCase))
+            !studyInstanceUid.Equals(requiredStudyInstanceUid.Trim(), StringComparison.OrdinalIgnoreCase))
```

Comment:
Is this only applicable for studyUID or for other UIDs too? StudyUID comes as part of request and others are from the dataset